### PR TITLE
implement Config class

### PR DESCRIPTION
**Pull request by tianyime**

### CODE REVIEW

File: src/common/config.h [hunk 1]
```diff
@@ -9,13 +10,29 @@ namespace raftcpp {
...
+        for (; pos != end; ++pos) {
+            config.push_back(Endpoint(pos->str()));
+        }
+        return config;
+    }
 
     Endpoint GetSelfEndpoint() const {}
 
-    std::vector<Endpoint> GetAllEndpoints() const {}
+    void push_back(const Endpoint &endpoint) {
```
Comment (jovany-wang) [inline src/common/config.h:27]: 1. `push_back` should be a private method.
2. It should be renamed to `PushBack()`.

File: src/common/config.h [hunk 2]
```diff
@@ -9,13 +10,29 @@ namespace raftcpp {
 
 class Config final {
     public:
-    static Config From(const std::string &config_str) { return Config(); }
+    static Config From(const std::string &config_str) {
+        static Config config;
```
Comment (jovany-wang) [inline src/common/config.h:14]: I think this `config` shouldn't be local static.

File: src/common/config.h [hunk 3]
```diff
@@ -9,13 +10,29 @@ namespace raftcpp {
 
 class Config final {
     public:
-    static Config From(const std::string &config_str) { return Config(); }
```
Comment (jovany-wang) [inline src/common/config.h:12]: The ctor of Config should be private or deleted.

**Review by jovany-wang — COMMENTED** [review]
Looks good overall.

File: src/common/config.h (hunk 1 above)
Comment (tianyime) [reply src/common/config.h]: Done

File: src/common/config.h (hunk 2 above)
Comment (tianyime) [reply src/common/config.h]: Done

File: src/common/config.h (hunk 3 above)
Comment (tianyime) [reply src/common/config.h:12]: Done

**Review by jovany-wang — APPROVED** [review]

**Review by jovany-wang — APPROVED** [review]

**Pull request merged**